### PR TITLE
.Net: added Functions.OpenApi.Extensions project

### LINKED ISSUE #4851: .Net: Create Functions.OpenAPI.Extensions project
As part of an offline conversation with @SergeyMenshykh, I got the recommendation to create an extensions package to host APIManifest Plugin implementation from #4662. The package may host other extensibility features, such as OpenAI plugin implementation at a later stage.

### PR DESCRIPTION
**Pull request by zengin**
### Motivation and Context
- fixes #4851

### Description

Adds an empty project to host OpenAPI plugin extensions e.g. API Manifest plugins (related PR: #4662)

### Contribution Checklist

- [x] The code builds clean without any errors or warnings
- [x] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [x] All unit tests pass, and I have added new tests where possible
- [x] I didn't break anyone :smile:

### CODE REVIEW

File: dotnet/SK-dotnet.sln [hunk 1]
```diff
@@ -470,6 +472,12 @@ Global
 		{87AB5AF5-5783-4372-9789-664895E0A2FF}.Publish|Any CPU.Build.0 = Debug|Any CPU
 		{87AB5AF5-5783-4372-9789-664895E0A2FF}.Release|Any CPU.ActiveCfg = Release|Any CPU
 		{87AB5AF5-5783-4372-9789-664895E0A2FF}.Release|Any CPU.Build.0 = Release|Any CPU
+		{95CAA25F-A0DE-4A5B-92BA-7D56C0E822A8}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
+		{95CAA25F-A0DE-4A5B-92BA-7D56C0E822A8}.Debug|Any CPU.Build.0 = Debug|Any CPU
+		{95CAA25F-A0DE-4A5B-92BA-7D56C0E822A8}.Publish|Any CPU.ActiveCfg = Debug|Any CPU
+		{95CAA25F-A0DE-4A5B-92BA-7D56C0E822A8}.Publish|Any CPU.Build.0 = Debug|Any CPU
```
Comment (SergeyMenshykh) [inline dotnet/SK-dotnet.sln:478]: The publish config is used by our release pipeline to identify projects to build, package and publish to the SK NuGet feed. We suggest removing the two "publish" lines and adding them later when the final code, which we would like to be publicly available, is in place.

Comment (zengin) [reply dotnet/SK-dotnet.sln]: makes sense, thanks for letting me know. Removed the publish config now.

**Review by SergeyMenshykh — APPROVED** [review]

**Comment by zengin** [comment]
@SergeyMenshykh, the .md link failures seem unrelated to my change: https://github.com/microsoft/semantic-kernel/actions/runs/7819466767/job/21331967577?pr=4867, how do you recommend we proceed?

**Comment by SergeyMenshykh** [comment]
> @SergeyMenshykh, the .md link failures seem unrelated to my change: https://github.com/microsoft/semantic-kernel/actions/runs/7819466767/job/21331967577?pr=4867, how do you recommend we proceed?

@zengin we are working on the fix at the moment. I'll let you know when we are done.

**Comment by SergeyMenshykh** [comment]
> > @SergeyMenshykh, the .md link failures seem unrelated to my change: https://github.com/microsoft/semantic-kernel/actions/runs/7819466767/job/21331967577?pr=4867, how do you recommend we proceed?
> 
> @zengin we are working on the fix at the moment. I'll let you know when we are done.

@zengin we applied a temp workaround so you can proceed with the PR.

**Review by markwallace-microsoft — APPROVED** [review]

**Pull request merged**